zhmcclient/_metrics.py

```diff
@@ -0,0 +1,390 @@
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+        """
+        result = self.session.post('/api/services/metrics/context',
+                                   body=properties)
+        # There should not be overlaps, but just in case there are, the
+        # returned props should overwrite the input props:
+#        print(result)
+#        props = properties.copy()
+#        props.update(result)
+        new_metrics_context = MetricsContext(self,
+                                             result['metrics-context-uri'],
+                                             None,
+                                             result)
+        self._metrics_contexts.append(new_metrics_context)
+        return new_metrics_context
+
+    @property
+    def client(self):
+        return self._client
+
+
+class MetricsContext(BaseResource):
+    """
+    Representation of a :term:`Metrics Context`.
+
+    Derived from :class:`~zhmcclient.BaseResource`; see there for common
+    methods and attributes.
+
+    Objects of this class are directly created by the user.
+    """
+
+    def __init__(self, manager, uri, name=None, properties=None):
+        # This function should not go into the docs.
+        #   manager (:class:`~zhmcclient.CpcManager`):
+        #     Manager object for this resource object.
+        #   uri (string):
+        #     Canonical URI path of the resource.
+        #   name (string):
+        #     Name of the resource.
+        #   properties (dict):
+        #     Properties to be set for this resource object. May be `None` or
+        #     empty.
+        if not isinstance(manager, MetricsContextManager):
+            raise AssertionError(
+                "MetricsContext init: Expected manager type %s, got %s" %
+                (MetricsContextManager, type(manager)))
+        super(MetricsContext, self).__init__(manager, uri, name, properties)
+
+    @logged_api_call
+    def get_metrics(self):
+        """
+        Start this CPC, using the HMC operation "Start CPC".
+
+        Returns:
+
+          :term:`json object`:
+
+            If `wait_for_completion` is `True`, returns None.
+
+            If `wait_for_completion` is `False`, returns a JSON object with a
+            member named ``job-uri``. The value of ``job-uri`` identifies the
+            job that was started, and can be used with the
+            :meth:`~zhmcclient.Session.query_job_status` method to determine
+            the status of the job and the result of the asynchronous HMC
+            operation, once the job has completed.
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+        """
+        result = self.manager.session.get(self.uri)
+        return result
```

Comment:
We should not hand back the result as a JSON object, but as a proper Python object (I would expect a list of `MetricsGroup` objects, because a metrics context stands for a list of metrics groups)